documentation/dataplane/dash-flow/DASH_FLOW_API_HLD.md

```diff
@@ -0,0 +1,982 @@
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_remove_flow_entry_fn)(
+        _In_ const sai_flow_entry_t *flow_entry);
+
+/**
+ * @brief Set attribute for dash_flow_flow_entry
+ *
+ * @param[in] flow_entry Entry
+ * @param[in] attr Attribute
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_set_flow_entry_attribute_fn)(
+        _In_ const sai_flow_entry_t *flow_entry,
+        _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief Get attribute for dash_flow_flow_entry
+ *
+ * @param[in] flow_entry Entry
+ * @param[in] attr_count Number of attributes
+ * @param[inout] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_get_flow_entry_attribute_fn)(
+        _In_ const sai_flow_entry_t *flow_entry,
+        _In_ uint32_t attr_count,
+        _Inout_ sai_attribute_t *attr_list);
+
+/**
+ * @brief Bulk create dash_flow_flow_entry
+ *
+ * @param[in] object_count Number of objects to create
+ * @param[in] flow_entry List of object to create
+ * @param[in] attr_count List of attr_count. Caller passes the number
+ *    of attribute for each object to create.
+ * @param[in] attr_list List of attributes for every object.
+ * @param[in] mode Bulk operation error handling mode.
+ * @param[out] object_statuses List of status for every object. Caller needs to
+ * allocate the buffer
+ *
+ * @return #SAI_STATUS_SUCCESS on success when all objects are created or
+ * #SAI_STATUS_FAILURE when any of the objects fails to create. When there is
+ * failure, Caller is expected to go through the list of returned statuses to
+ * find out which fails and which succeeds.
+ */
+typedef sai_status_t (*sai_bulk_create_flow_entry_fn)(
+        _In_ uint32_t object_count,
+        _In_ const sai_flow_entry_t *flow_entry,
+        _In_ const uint32_t *attr_count,
+        _In_ const sai_attribute_t **attr_list,
+        _In_ sai_bulk_op_error_mode_t mode,
+        _Out_ sai_status_t *object_statuses);
+
+/**
+ * @brief Bulk remove dash_flow_flow_entry
+ *
+ * @param[in] object_count Number of objects to remove
+ * @param[in] flow_entry List of objects to remove
+ * @param[in] mode Bulk operation error handling mode.
+ * @param[out] object_statuses List of status for every object. Caller needs to
+ * allocate the buffer
+ *
+ * @return #SAI_STATUS_SUCCESS on success when all objects are removed or
+ * #SAI_STATUS_FAILURE when any of the objects fails to remove. When there is
+ * failure, Caller is expected to go through the list of returned statuses to
+ * find out which fails and which succeeds.
+ */
+typedef sai_status_t (*sai_bulk_remove_flow_entry_fn)(
+        _In_ uint32_t object_count,
+        _In_ const sai_flow_entry_t *flow_entry,
+        _In_ sai_bulk_op_error_mode_t mode,
+        _Out_ sai_status_t *object_statuses);
+
+```
+
+### Bulk operation
+
```

Comment:
here, we need to explain why we need to have this filter / the design of the bulk get function.